src/JustEat.StatsD/StatsDMessageFormatter.cs

```diff
@@ -5,6 +5,71 @@
 
 namespace JustEat.StatsD
 {
+    internal static class StringBuilderCache
```

Comment:
Was this just lifted wholesale out of the corefx repo? https://github.com/dotnet/coreclr/blob/3bfaf1f10dfc60972ee4e6116eb8607ecd1554df/src/System.Private.CoreLib/shared/System/Text/StringBuilderCache.cs

I don't think we should be really adding our own copies of internal framework types.

Reply:
wow did not know it was in corefx, stolen from techempower benchmarks for .net middleware https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/CSharp/aspnetcore/PlatformBenchmarks/Utilities/StringBuilderCache.cs

Reply:
If you’re taking code for other sources, it should be as-is with copyright headers intact anyway? I’m not a lawyer, but there’s legal considerations in play if it’s from elsewhere.

Reply:
corefx is [MIT Licensed](https://github.com/dotnet/corefx/blob/ba505f52eb415364c1fd4f52ac00c51f37b8d125/LICENSE.TXT#L1), and we're Apache 2.0, so I'm not sure how they interact with each other either.